File: julia/002b.jl
LIMIT = 4*1000*1000
sum = 0

nextfib = (() ->
    begin
        a, b = 0, 1

        function()
            a, b = b, a + b
            b
        end
    end)()

while true
    f = nextfib()
    if f > LIMIT
        break
    end

    if f % 2 == 0
        sum += f
    end
end

println(sum)

File: r/002b.r
LIMIT <- 4*1000*1000
sum <- 0

nextfib <- (function() {
    a <- 0
    b <- 1

    function() {
        tmp <- a
        a   <<- b
        b   <<- tmp + b

        b
    }
})()

while (TRUE) {
    f <- nextfib()

    if (f > LIMIT) {
        break
    }

    if (f %% 2 == 0) {
        sum <- sum + f
    }
}

print(sum)
